chroma_core/services/job_scheduler/job_scheduler.py

```diff
@@ -1745,3 +1755,104 @@ def update_lnet_configuration(self, lnet_configuration_list):
     @property
     def CommandPlan(self):
         return CommandPlan(self._lock_cache, self._job_collection)
+
+    def configure_stratagem(self, stratagem_data):
+        with self._lock:
+            configuration_data = {
+                "state": "unconfigured",
+                "filesystem_id": stratagem_data.get("filesystem"),
+                "interval": stratagem_data.get("interval"),
+                "report_duration": stratagem_data.get("report_duration"),
+                "purge_duration": stratagem_data.get("purge_duration"),
+            }
+
+            # The filesystem_id may come in as the fs name or the fs id. In terms of storing information in the database, the fs id should always be used.
+            fs_identifier = str(configuration_data.get("filesystem_id"))
+            fs_id = get_fs_id_from_identifier(fs_identifier)
+
+            if not fs_id:
+                from tastypie.exceptions import NotFound
+                raise NotFound("No matching filesystem for {}".format(fs_identifier))
```

Comment:
This should be a generic error, not a Tastypie one. What happens if this is called outside the API?